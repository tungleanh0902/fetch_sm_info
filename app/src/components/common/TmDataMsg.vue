<template>
  <div class="tm-data-msg">
    <div class="tm-data-msg__icon" :class="{ success: success }">
      <i
        :style="`background: ${iconColor}`"
        :class="spinnerClass"
        class="material-icons notranslate"
        >{{ icon }}</i
      >
    </div>
    <div class="tm-data-msg__text">
      <h2 class="tm-data-msg__title">
        <slot name="title">
          {{ title }}
        </slot>
      </h2>
      <h3 class="tm-data-msg__subtitle">
        <slot name="subtitle">
          {{ subtitle }}
        </slot>
      </h3>
    </div>
  </div>
</template>

<script>
export default {
  name: `tm-data-msg`,
  props: {
    title: {
      type: String,
      default: null,
    },
    subtitle: {
      type: String,
      default: null,
    },
    icon: {
      type: String,
      default: null,
    },
    spin: {
      type: Boolean,
      default: false,
    },
    success: {
      type: Boolean,
      default: false,
    },
    iconColor: {
      type: String,
      default: `var(--warning)`,
    },
  },
  computed: {
    spinnerClass({ spin } = this) {
      return spin ? `fa-spin` : ``
    },
  },
}
</script>

<style>
.tm-data-msg {
  padding: 2rem;
  display: flex;
  align-items: center;
  border-radius: 0.25rem;
  background: var(--app-fg);
  border: 2px solid var(--app-fg-hover);
}

.tm-data-msg__icon {
  margin-right: 1.5rem;
}

.tm-data-msg__icon i.material-icons {
  font-size: 2rem;
  color: var(--menu-bright);
  background: var(--warning);
  padding: 0.5rem;
  border-radius: 50%;
}

.tm-data-msg__icon.success i.material-icons {
  background: var(--success);
}

.tm-data-msg__title {
  font-weight: 400;
  color: var(--bright);
  font-size: var(--h2);
}

.tm-data-msg__subtitle {
  color: var(--txt);
  font-size: 1rem;
  word-break: break-word;
}

.fa-spin {
  animation: spin 1400ms linear infinite;
}

@keyframes spin {
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
}

@media screen and (max-width: 767px) {
  .tm-data-msg {
    padding: 2rem 1rem;
    margin: 1rem;
  }
}

@media screen and (max-width: 1023px) {
  .tm-data-msg {
    display: block;
    padding: 2rem;
  }

  .tm-data-msg__icon {
    margin-bottom: 1rem;
  }
}
</style>
