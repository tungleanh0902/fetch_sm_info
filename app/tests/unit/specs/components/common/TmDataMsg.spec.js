import { mount } from "@vue/test-utils"
import TmDataMsg from "common/TmDataMsg"

describe(`TmDataMsg`, () => {
  let wrapper

  const propsData = {
    title: `sOmEtHiNg Is HaPpEnIng!!!`,
    subtitle: `Oh my! What could it be?`,
    icon: `help`,
    spin: false,
  }

  beforeEach(() => {
    wrapper = mount(TmDataMsg, { propsData })
  })

  it(`has a title from props`, () => {
    expect(wrapper.vm.title).toBe(`sOmEtHiNg Is HaPpEnIng!!!`)
  })

  it(`has the expected html structure`, () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it(`has an icon`, () => {
    expect(wrapper.find(`.tm-data-msg__icon i.material-icons`).text()).toBe(
      `help`
    )
  })

  it(`has shows a spinner`, () => {
    wrapper.setProps({ spin: true })
    expect(wrapper.contains(`.fa-spin`)).toBe(true)
  })

  it(`doesn't have a spinning icon`, () => {
    expect(
      wrapper.find(`.tm-data-msg__icon`).contains(`i.material-icons.fa-spin`)
    ).toBe(false)
  })

  it(`has a subtitle`, () => {
    expect(wrapper.find(`.tm-data-msg__subtitle`).text()).toBe(
      `Oh my! What could it be?`
    )
  })
})
