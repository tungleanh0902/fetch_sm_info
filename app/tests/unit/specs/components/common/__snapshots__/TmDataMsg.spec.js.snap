// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TmDataMsg has the expected html structure 1`] = `
<div
  class="tm-data-msg"
>
  <div
    class="tm-data-msg__icon"
  >
    <i
      class="material-icons notranslate"
    >
      help
    </i>
  </div>
   
  <div
    class="tm-data-msg__text"
  >
    <h2
      class="tm-data-msg__title"
    >
      
        sOmEtHiNg Is HaPpEnIng!!!
      
    </h2>
     
    <h3
      class="tm-data-msg__subtitle"
    >
      
        Oh my! What could it be?
      
    </h3>
  </div>
</div>
`;
